libkbfs/journal_server_test.go

```diff
@@ -72,6 +75,148 @@ func teardownJournalServerTest(
 	assert.NoError(t, err)
 }
 
+type quotaBlockServer struct {
+	BlockServer
+
+	quotaInfoLock sync.Mutex
+	quotaInfo     kbfsblock.UserQuotaInfo
+}
+
+func (qbs *quotaBlockServer) setUserQuotaInfo(
+	remoteUsageBytes, limitBytes int64) {
+	qbs.quotaInfoLock.Lock()
+	defer qbs.quotaInfoLock.Unlock()
+	qbs.quotaInfo.Limit = limitBytes
+	qbs.quotaInfo.Total = &kbfsblock.UsageStat{
+		Bytes: map[kbfsblock.UsageType]int64{
+			kbfsblock.UsageWrite: remoteUsageBytes,
+		},
+	}
+}
+
+func (qbs *quotaBlockServer) GetUserQuotaInfo(ctx context.Context) (
+	info *kbfsblock.UserQuotaInfo, err error) {
+	qbs.quotaInfoLock.Lock()
+	defer qbs.quotaInfoLock.Unlock()
+	infoCopy := qbs.quotaInfo
+	return &infoCopy, nil
+}
+
+func TestJournalServerEstimateQuotaUsage(t *testing.T) {
+	tempdir, ctx, cancel, config, jServer := setupJournalServerTest(t)
+	defer teardownJournalServerTest(t, tempdir, ctx, cancel, config)
+
+	qbs := &quotaBlockServer{BlockServer: config.BlockServer()}
+	config.SetBlockServer(qbs)
+
+	// Check initial state.
+	localUsageBytes, remoteUsageBytes, limitBytes :=
+		jServer.estimateQuotaUsage(ctx)
+	require.Equal(t, int64(0), localUsageBytes)
+	require.Equal(t, int64(0), remoteUsageBytes)
+	require.Equal(t, int64(math.MaxInt64), limitBytes)
+
+	// Set quota usage since there's a background task to retrieve
+	// the quota usage, which will return 0 for the limit.
+	qbs.setUserQuotaInfo(100, 1000)
+	// Force quotaUsage to update.
+	_, _, _, err := jServer.quotaUsage.Get(ctx, 0, 0)
+	require.NoError(t, err)
+
+	tlfID1 := tlf.FakeID(1, false)
+	err = jServer.Enable(ctx, tlfID1, TLFJournalBackgroundWorkPaused)
+	require.NoError(t, err)
+
+	tlfID2 := tlf.FakeID(2, false)
+	err = jServer.Enable(ctx, tlfID2, TLFJournalBackgroundWorkPaused)
+	require.NoError(t, err)
+
+	blockServer := config.BlockServer()
+
+	h, err := ParseTlfHandle(
+		ctx, config.KBPKI(), "test_user1,test_user2", false)
+	require.NoError(t, err)
+	uid1 := h.ResolvedWriters()[0]
+
+	// Put blocks to TLFs 1 and 2.
+
+	bCtx := kbfsblock.MakeFirstContext(uid1, keybase1.BlockType_DATA)
+	data := []byte{1, 2, 3, 4}
+	bID, err := kbfsblock.MakePermanentID(data)
+	require.NoError(t, err)
+	serverHalf, err := kbfscrypto.MakeRandomBlockCryptKeyServerHalf()
+	require.NoError(t, err)
+	err = blockServer.Put(ctx, tlfID1, bID, bCtx, data, serverHalf)
+	require.NoError(t, err)
+	err = blockServer.Put(ctx, tlfID2, bID, bCtx, data, serverHalf)
+	require.NoError(t, err)
+
+	localUsageBytes, remoteUsageBytes, limitBytes =
+		jServer.estimateQuotaUsage(ctx)
```

Comment:
How can you be sure the background quota-fetching goroutine has finished by the time this runs?